SCF-Session-6/polynomial/Poly.java

```diff
@@ -0,0 +1,225 @@
+package polynomial;
+
+/**
+ * immutable class Poly which uses an array to represent single variable polynomial
+ * @author Ishita_Mathur
+ *
+ */
+public final class Poly 
+{
+    private int[][] polynomial;
+    private static final int COEFFICIENT = 0;
+    private static final int POWER = 1;
+    private static final int COLUMN = 2;
+    
+    public int[][] getPolynomial() 
+    {
+        int[][] newPolynomial = new int[size()][COLUMN];
+        
+        for(int i = 0; i < size(); i++)
+        {
+            newPolynomial[i][COEFFICIENT] = polynomial[i][COEFFICIENT];
+            newPolynomial[i][POWER] = polynomial[i][POWER];
+        }
+        
+        return newPolynomial;
+    }
+    
+    
+    /**
+     * constructor of the class
+     * @param inputPolynomial a 2-d array having coefficients and corresponding powers,
+     * assumes coefficient will be integer
+     */
+    public Poly(int[][] inputPolynomial)
```

Comment:
Constructor should be defined before any method of the class, just after variable declaration section.

Reply:
done